src/graphnet/data/sqlite_dataset.py

```diff
@@ -161,12 +187,25 @@ def _create_graph(self, features, truth):
             data = np.array([]).reshape((0, len(self._features) - 1))
 
         # Construct graph data object
-        x = torch.tensor(data, dtype=self._dtype)
-        n_pulses = torch.tensor(len(x), dtype=torch.int32)
-        graph = Data(
-            x=x,
-            edge_index= None
-        )
+        if self._node_representation.lower() == 'pulse':
+            x = torch.tensor(data, dtype=self._dtype)
+            n_pulses = torch.tensor(len(x), dtype=torch.int32)
+            graph = Data(
+                x=x,
+                edge_index= None
+            )
+        elif self._node_representation.lower() == 'dom':
+            x = torch.tensor(data, dtype=self._dtype)
+            n_pulses = torch.tensor(len(x), dtype=torch.int32)
+            graph = Data(
+                x=x,
+                edge_index= None
+            )
+            graph = self._make_dom_wise_representation(graph)
+        else:
+            print('WARNING: node representation %s not recognized!'%self._node_representation)
+            
+        
```

Comment:
DRY. Also, the check for `node_representation in ['pulse', 'node']` should probably be done in the constructor, such that we can assume that it has a standard value in all other methods. I'll assume that here to simplify.
```suggestion
        x = torch.tensor(data, dtype=self._dtype)
        n_pulses = torch.tensor(len(x), dtype=torch.int32)
        graph = Data(
            x=x,
            edge_index= None
        )
            
        if self._node_representation == 'dom':
            graph = self._make_dom_wise_representation(graph)
```